tools/utils/policy_test

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+import os
+import argparse
+import json
+import yaml
+import tests.common
+import c7n.filters.core
+import c7n.testing
+import c7n.utils
+
+def patch(replay):
+    process_set = c7n.filters.core.Not.process_set
+    def _process_set(self, resources, event):
+        results = process_set(self, resources, event)
+        if results:
+            key = None
+            keys = results[0].keys()
+            if 'Arn' in keys:
+                key = 'Arn'
+            elif 'GroupId' in keys:
+                key = 'GroupId'
+            results = sorted(results, key=lambda x: x[key])
+        return results
+    c7n.filters.core.Not.process_set = _process_set
+
+    if replay:
+        c7n.policy.Policy._write_file = lambda self, rel_path, value: None
```

Comment:
Some messy monkeypatching here to prevent `replay` mode from writing out any files...

Reply:
yeah, the monkey here is fine, I wish we had a clean way of disabling this, but blob store and write file got wired in very early and hasn't been revisited 🤷‍♂